viadot/sources/sharepoint.py

```diff
@@ -65,3 +69,29 @@ def download_file(
             url=download_from_path,
             filename=download_to_path,
         )
+
+    def download_file_from_url(self, rel_file_url, download_folder: str = "./") -> None:
+        """Function to download a file from sharepoint, given its URL
+        Args:
+            abs_url (str): URL to sharepoint website (e.g.: {tenant_name}.sharepoint.com)
+            rel_file_url (str): Relative URL to the file (e.g.: /sites/{directory}/Shared%20Documents/Downloads/)
+            credentials (str,str): Tuple with user and password (for now)
+            download_folder (str): Path where the file will be downloaded to
+
+        """
+        abs_url = self.credentials["site"]
+        rel_file_url = rel_file_url or self.url
+        user_name = self.credentials["username"]
+        user_password = self.credentials["password"]
```

Comment:
In the final version, it will be client_id and client_secret. It should be initialized in the `__init__` method not in the `doenload_...` method.